### PR TITLE
ci: build deps from source on openSUSE

### PR DESCRIPTION
**Pull request by dbolduc**
Silences #13921 although we may want to revert this change if the packaging is fixed upstream.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13923)

### CODE REVIEW

File: ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile [hunk 1]
```diff
@@ -50,6 +49,103 @@ ENV PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/usr/local/lib64/pkgconfig
 ENV PATH=/usr/local/bin:${PATH}
 # ```
 
+# #### Abseil
```
Comment (coryan) [inline ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile:52]: I would continue to pull this from the packages.

File: ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile [hunk 2]
```diff
@@ -50,6 +49,103 @@ ENV PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/usr/local/lib64/pkgconfig
...
+      -DABSL_BUILD_TESTING=OFF \
+      -DABSL_PROPAGATE_CXX_STD=ON \
+      -DBUILD_SHARED_LIBS=yes \
+      -S . -B cmake-out && \
+    cmake --build cmake-out -- -j ${NCPU:-4} && \
+    cmake --build cmake-out --target install -- -j ${NCPU:-4} && \
+    ldconfig
+# ```
+
+# #### RE2
```
Comment (coryan) [inline ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile:73]: I think you can continue using RE2 from the packages.

File: ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile [hunk 3]
```diff
@@ -50,6 +49,103 @@ ENV PKG_CONFIG_PATH=/usr/local/lib/pkgconfig:/usr/local/lib64/pkgconfig
...
+# distributes c-ares-1.9. Manually install a newer version:
+
+# ```bash
+WORKDIR /var/tmp/build/c-ares
+RUN curl -fsSL https://github.com/c-ares/c-ares/archive/cares-1_14_0.tar.gz | \
+    tar -xzf - --strip-components=1 && \
+    ./buildconf && ./configure && make -j ${NCPU:-4} && \
+    make install && \
+    ldconfig
+# ```
```
Comment (coryan) [inline ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile:121]: I got c-ares 1.19 with `zypper install c-ares-devel`.

Comment (dbolduc) [reply ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile]: Reverted to using the package. Thanks.

File: ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile (hunk 1 above)
Comment (dbolduc) [reply ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile]: Done. Thanks.

File: ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile (hunk 2 above)
Comment (dbolduc) [reply ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile:52]: My testing says no:

```
/usr/lib64/gcc/x86_64-suse-linux/8/../../../../x86_64-suse-linux/bin/ld: /usr/local/lib/libgrpc.so.39.0.0: undefined reference to `re2::RE2::RE2(absl::lts_20240116::string_view, re2::RE2::Options const&)'
/usr/lib64/gcc/x86_64-suse-linux/8/../../../../x86_64-suse-linux/bin/ld: /usr/local/lib/libgrpc.so.39.0.0: undefined reference to `re2::RE2::GlobalReplace(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, re2::RE2 const&, absl::lts_20240116::string_view)'
collect2: error: ld returned 1 exit status
```

Comment (coryan) [reply ci/cloudbuild/dockerfiles/demo-opensuse-leap.Dockerfile:52]: Ah, RE2 also uses Abseil, oh the joys.

**Review by coryan — APPROVED** [review]

**Pull request merged**